### ISSUE TITLE
Background tasks don't disappear even when cancelled

### ISSUE THREAD

**Issue by alejandro-fiel-stori** [dbeaver/dbeaver#34658]
### Description

This is an example of my "Background tasks" after using Dbeaver on Redshift for some minutes
![image](https://github.com/dbeaver/dbeaver/assets/153534002/79bd3227-20af-48c4-bef3-3c3c962fb4c4)
Nothing of those are shown on Redshift, so, they are all canceled, why they don't disappear? Is Dbeaver trying to do something with those queries?

### DBeaver Version

Community 24.1.1

### Operating System

macOS 14

### Database and driver

Redshift
com.amazon.redshift:redshift-jdbc42:RELEASE 2.1.0.29

### Steps to reproduce

Execute some queries, cancel some queries, they will start piling there

**Comment by uslss** [comment]
Thanks for the report

for me issue is reproducible e.g. on Oracle on the attempt to open SYS schema while I have issues with connection. Also error below appears in the log
![image](https://github.com/dbeaver/dbeaver/assets/31996417/2e0c2804-2651-40a6-b566-84127e0cb780)

```
eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Framework arguments:  -eclipse.keyring C:\Users\Dziyana\AppData\Roaming\DBeaverData\secure\secure_storage
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Mon Jul 08 14:05:07 CEST 2024
Couldn't close JDBC connection: timeout

org.jkiss.dbeaver.model.exec.DBCException: Couldn't close JDBC connection: timeout
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCConnectionImpl.cancelBlock(JDBCConnectionImpl.java:713)
	at org.jkiss.dbeaver.model.runtime.BlockCanceler.cancelBlock(BlockCanceler.java:36)
	at org.jkiss.dbeaver.model.runtime.AbstractJob$JobCanceler.run(AbstractJob.java:254)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

```

**Comment by serjiokov** [comment]
Can not reproduce

**Comment by alejandro-fiel-stori** [comment]
It was solved in the last version (24.1.3)
Thank you!

**Comment by E1izabeth** [comment]
@alejandrofielm thank you for update! I'm glad the issue is solved.
